service/task_manager_module.cc

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "locator/tablets.hh"
+#include "replica/database.hh"
+#include "service/storage_service.hh"
+#include "service/task_manager_module.hh"
+#include "tasks/task_handler.hh"
+
+namespace service {
+
+tasks::task_manager::task_group tablet_virtual_task::get_group() const noexcept {
+    return tasks::task_manager::task_group::tablets_group;
+}
+
+future<std::optional<tasks::task_manager::virtual_task_hint>> tablet_virtual_task::contains(tasks::task_id task_id) const {
+    auto tables = get_table_ids();
+    for (auto table : tables) {
+        bool has_task = false;
+        auto& tmap = _ss.get_token_metadata().tablets().get_tablet_map(table);
+        co_await tmap.for_each_tablet([&] (locator::tablet_id tid, const locator::tablet_info& info) {
+            has_task = has_task || (info.repair_task_info.is_valid() && info.repair_task_info.tablet_task_id.uuid() == task_id.uuid());
+            return make_ready_future();
+        });
+
+        if (has_task) {
+            co_return tasks::task_manager::virtual_task_hint{{TABLE_ID, table.to_sstring()}};
+        }
+    }
+    co_return std::nullopt;
+}
+
+future<tasks::is_abortable> tablet_virtual_task::is_abortable(tasks::task_manager::virtual_task_hint hint) const {
+    return make_ready_future<tasks::is_abortable>(tasks::is_abortable::yes);
+}
+
+future<std::optional<tasks::task_status>> tablet_virtual_task::get_status(tasks::task_id id, tasks::task_manager::virtual_task_hint hint) {
+    auto table = table_id{utils::UUID{hint.at(TABLE_ID)}};
```

Comment:
What guarantees TABLE_ID is in the hint? (Perhaps the caller always provide one, but still it is better to check)

If we have

struct virtual_task_hint {
std::optional<table_id> table_id;
std::optional<other_thing_for_the_hint> foo;
}

we could do

if (hint.table_id) {
}

Reply:
I changed it to `on_internal_error` if TABLE_ID isn't there. This situation should not occur